E3Next/Processors/Bots.cs

```diff
@@ -770,6 +772,48 @@ public void Trade(string name)
             string command = "/notify TradeWnd TRDW_Trade_Button leftmouseup";
 			PubServer.AddTopicMessage("OnCommand-" + name, $"{E3.CurrentName}:{false}:{command}");
 		}
+
+        public void BroadcastLootingCorpse(int corpseId, bool looting)
+        {
+            PubServer.AddTopicMessage("Looting", $"{corpseId}:{looting}");
+        }
+
+        Dictionary<string, SharedNumericDataInt32> _lootCollection = new Dictionary<string, SharedNumericDataInt32>();
+        public bool IsGroupLooting(int corpseId)
+        {
+            string keyToUse = "Looting";
+            var allTopics = NetMQServer.SharedDataClient.TopicUpdates;
+
+            foreach (var userTopics in allTopics.Where(x => x.Key != E3.CurrentName))
+            {
+                if (!userTopics.Value.ContainsKey(keyToUse))
+                {
+                    continue;
+                }
+                var entry = userTopics.Value[keyToUse];
+                if (!_lootCollection.ContainsKey(userTopics.Key))
+                {
+                    _lootCollection.Add(userTopics.Key, new SharedNumericDataInt32 { Data = 0 });
+                }
+                var sharedInfo = _lootCollection[userTopics.Key];
+
+                lock (entry)
+                {
+                    if (entry.LastUpdate > sharedInfo.LastUpdate)
+                    {
+                        var split = entry.Data.Split(':');
+                        if (int.TryParse(split[0], out var corpse) && bool.TryParse(split[1], out var looting))
+                        {
+                            sharedInfo.Data = looting ? corpse : 0;
```

Comment:
wonder if changing this data type to an LRU (least recently used) type of say, 50, and then keep that updated so that it isn't only during the process of looting but making it so that if they touch the corpse, even if the corpse still persists they will ignore it.

Reply:
will think on it more tomorrow when its not 1am :)

Reply:
or possibly the circular buffer that is in core.cs, but again, tomorrow

Reply:
I can change it to use the circular buffer in core. I usually have it delete anything I'm not looting so the corpses don't stay around... Were you thinking it should be a circular buffer per character or a single circular buffer for all looted corpse ids?

Reply:
(correct me if i'm wrong, this is from my memory of last night) 
My thought is Me.Looting only has the current ID, so it will have either an ID or Zero? (seems to be based off the PercentHP code I made which was a good idea) and people also use loot settings that only loot the things they care about. So by keeping a small list instead of just an ID, would allow that form of looting to also work the same way.  Would have to create a new object to contain each users circular buffer.. which is actually just an array with logic in it to act like a circular buffer :)

Reply:
and to be clear, I meant a circular buffer for each user which would have say.. their last 10 loot ID's in them?

Reply:
Yea, right now it just has a single corpse id if a character is looting, 0 if they aren't, I'll change it to use circular buffers :), thanks!

Reply:
Updated!